client/src/TeacherMoments.js

```diff
@@ -0,0 +1,431 @@
+}
+
+
+
+function percentage(statsForSessions, filterFn) {
+  return Math.round(100 * (statsForSessions.filter(filterFn).length / statsForSessions.length)) + '%';
+}
+
+
+
+
+
+// Decide what analysis to do
+class TeacherMoments extends Component {
+  constructor(props) {
+    super(props);
+    const analysisTuple = _.last(_.entries(Analyses));
+    const [key, analysis] = analysisTuple;
+    this.state = {
+      key: key, 
+      analysis: analysis,
+      location: '/teachermoments/turner?playtest20180124',
+      token: this.props.token,
+      email: this.props.email
+    };
+  }
+
+  onAnalysisChanged(e, index, targetAnalysisKey) {
+    const analysisTuple = _.find(_.entries(Analyses), pair => pair[0] === targetAnalysisKey);
+    const [key, analysis] = analysisTuple;
+    this.setState({key, analysis});
+  }
+
+  render() {
+    const currentKey = this.state.key;
+    const currentAnalysis = this.state.analysis;
+    const {filter, dataSet} = currentAnalysis;
+    return (
+      <MuiThemeProvider>
+        <div className="TeacherMoments">
+          <div className="TeacherMoments-header">
+            <img src={logo} className="TeacherMoments-logo" alt="logo" />
+            {this.renderSelect(currentKey)}
+          </div>
+          <Analysis
+            key={currentKey}
+            analysisKey={currentKey}
+            filter={filter}
+            dataSet={dataSet} 
+            location = {this.state.location}
+            token = {this.state.token}
+            email = {this.state.email}/>
+        </div>
+      </MuiThemeProvider>
+    );
+  }
+
+  renderSelect(currentKey) {
+    return (
+      <SelectField
+        style={{width: 'auto'}}
+        floatingLabelText="Analysis"
+        value={currentKey}
+        onChange={this.onAnalysisChanged.bind(this)}
+      >
+        {_.entries(Analyses).map((analysisTuple) => {
+          const [key, analysis] = analysisTuple;
+          const {description} = analysis;
+          return <MenuItem
+            key={key}
+            value={key}
+            primaryText={description} />;
+        })}
+      </SelectField>
+    );
+  }
+}
+
+
+
+
+class Analysis extends Component {
+  propTypes: {
+    analysisKey: React.PropTypes.string.isRequired,
+    filter: React.PropTypes.func.isRequired,
+    location: React.PropTypes.string.isRequired,
+    dataSet: {
+      db: React.PropTypes.string.isRequired,
+      s3: React.PropTypes.string.isRequired
+    }
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      json: null,
+      location: this.props.location,
+      token: this.props.token,
+      email: this.props.email
+    };
+  }
+
+  componentDidMount() {
+    const location = this.state.location;
+    const token = this.state.token;
+    const email = this.state.email;
+    fetch('/server/research/data', {
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'x-teachermoments-location': location,
```

Comment:
oh this got hidden since I renamed TeacherMoments.js

This makes sense. It sounds like a need a function that queries the token against a table that says who is authorized to see what data. I'll work on that!

Reply:
Actually, when you say "The server code is deciding what the user can see and limiting it to just that playtest", you're referring to me hard coding the location variable in database.js right?

I'm thinking I will still need the location in the header to communicate what data to query for in databse.js. I think I would need to use the token to confirm that this user can query for data at a specific location. That could probably be something like:
```
checkAccess(token, location)
  .then ( fetch with just the location header)
  .catch( console.log("you are not authorized to access this data"));
```
And so yeah, I shouldnt need the email once I have the token

Reply:
Cool, this is a great explanation, I think I'm following this.

So to show the user data, one thing the server needs to know is: "what set of data should I should them?"  And there's two parts: 1) what dataset the user wants, and 2) what they should be authorized to see.

When you said:
> It sounds like a need a function that queries the token against a table that says who is authorized to see what data. I'll work on that!
That's a great approach to this!  And I think what you'll want to do for sure soon.

But I was suggesting that for part 1) you just hard code that on the server so the only thing users can ask for is data for `?KevinTesting20180319` or whatever.  And then in this pull request all you need to do is 2) lookup to see if the user's token is valid and they have researcher access.  The downside to this is that in the short term all researchers can only access that same set of data you hardcoded.

Your suggestion is better overall for sure, so my comment here is just a suggestion that maybe you ship the hardcoded "what dataset you want" for now, and all you have to do is do the authorization check.  If you want to do both at once, and that seems better or simpler for any reason that's great too!

Reply:
gotcha! I think it makes sense to do just the hardcoded version first to finish off this pull request :)